File: app/src/main/java/com/alwaystinkering/sandbot/model/pattern/FileType.java
package com.alwaystinkering.sandbot.model.pattern;

public enum FileType {
    PARAM("param"),
    GCODE("gcode"),
    THETA_RHO("thr"),
    SEQUENCE("seq"),
    UNKNOWN("");

    FileType(String extension) {
        this.extension = extension;
    }

    private String extension;

    public String getExtension() {
        return extension;
    }

    public static FileType fromExtension(String extension) {
        for (FileType f : values()) {
            if (extension.equals(f.getExtension())) {
                return f;
            }
        }
        return UNKNOWN;
    }

}

File: app/src/main/java/com/alwaystinkering/sandbot/model/pattern/FileType.kt
package com.alwaystinkering.sandbot.model.pattern

enum class FileType constructor(val extension: String) {
    PARAM("param"),
//    GCODE("gcode"),
    THETA_RHO("thr"),
    SEQUENCE("seq");

    companion object {

        fun fromExtension(extension: String): FileType {
            for (f in values()) {
                if (extension == f.extension) {
                    return f
                }
            }
            return PARAM
        }
    }

}
